src/ensembl/src/content/app/browser/browser-nav/BrowserNavBar.tsx

```diff
@@ -1,50 +1,99 @@
-import React, { FunctionComponent } from 'react';
+import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
 import classNames from 'classnames';
 
-import { browserNavConfig, BrowserNavItem } from '../browserConfig';
+import BrowserNavIcon from './BrowserNavIcon';
+import BrowserRegionEditor from '../browser-region-editor/BrowserRegionEditor';
+import BrowserRegionField from '../browser-region-field/BrowserRegionField';
 
 import { RootState } from 'src/store';
-import { getBrowserNavStates } from '../browserSelectors';
+import { browserNavConfig, BrowserNavItem } from '../browserConfig';
+import {
+  getBrowserNavStates,
+  getChrLocation,
+  getRegionEditorActive,
+  getRegionFieldActive
+} from '../browserSelectors';
+import {
+  toggleRegionEditorActive,
+  toggleRegionFieldActive
+} from '../browserActions';
 import { getIsTrackPanelOpened } from '../track-panel/trackPanelSelectors';
-import { BrowserNavStates } from '../browserState';
-
-import BrowserNavIcon from './BrowserNavIcon';
+import { BrowserNavStates, ChrLocation } from '../browserState';
+import { getGenomeKaryotypes } from 'src/genome/genomeSelectors';
+import { GenomeKaryotype } from 'src/genome/genomeTypes';
 
 import styles from './BrowserNavBar.scss';
 
-type StateProps = {
+type BrowserNavBarProps = {
   browserNavStates: BrowserNavStates;
+  chrLocation: ChrLocation | null;
+  genomeKaryotypes: GenomeKaryotype[] | null;
   isTrackPanelOpened: boolean;
+  regionEditorActive: boolean;
+  regionFieldActive: boolean;
+  toggleRegionEditorActive: (regionEditorActive: boolean) => void;
+  toggleRegionFieldActive: (regionFieldActive: boolean) => void;
 };
 
-type BrowserNavBarProps = StateProps;
+export const BrowserNavBar = (props: BrowserNavBarProps) => {
+  // the region editor and field style should be reset so that it won't be opaque when nav bar is opened again
+  useEffect(
+    () => () => {
+      props.toggleRegionEditorActive(false);
+      props.toggleRegionFieldActive(false);
+    },
+    []
+  );
+
+  const shouldBeEnabled = (index: number) => {
+    const { browserNavStates, regionEditorActive, regionFieldActive } = props;
+    const maxState = browserNavStates[index];
```

Comment:
What does `maxState` mean here?

Reply:
It is a value that is returned from the genome browser when a maximum is hit on either of the navigation directions i.e. top, bottom, left and right. I just couldn't think of a better variable name for this when I wrote this.